### DISCUSSION TITLE
Kubectl in rancher/k3s Docker image doesn't work

### DISCUSSION THREAD

**Discussion by amcinnes** [k3s-io/k3s#8715] [General]
**Environmental Info:**
K3s Version: 
```
$ docker run --rm rancher/k3s@sha256:5f7b660e6f2a6dd712350a8f1fd5ad6beafd62a06df5e2f46e432e6f151652fc -v
k3s version v1.25.14+k3s1 (3624e6c4)
go version go1.20.8
```

Node(s) CPU architecture, OS, and Version: 
```
$ uname -a
Linux angus-laptop 6.5.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Tue, 10 Oct 2023 21:10:21 +0000 x86_64 GNU/Linux
```

Cluster Configuration: N/A

**Describe the bug:**

When I try to run `kubectl` as the command of the `rancher/k3s` docker image, I get this error message:
```
$ docker run --rm rancher/k3s@sha256:5f7b660e6f2a6dd712350a8f1fd5ad6beafd62a06df5e2f46e432e6f151652fc kubectl
error: unknown command "kubectl" for "kubectl"
```

**Steps To Reproduce:**

See above.

**Expected behavior:**
It should run kubectl. With no further arguments it should output something like this:
```
$ docker run --rm rancher/k3s@sha256:5f7b660e6f2a6dd712350a8f1fd5ad6beafd62a06df5e2f46e432e6f151652fc kubectl
kubectl controls the Kubernetes cluster manager.

 Find more information at: https://kubernetes.io/docs/reference/kubectl/

Basic Commands (Beginner):
  create          Create a resource from a file or from stdin
  expose          Take a replication controller, service, deployment or pod and expose it as a new Kubernetes service
[etc]
```

**Actual behavior:**

See above.

**Additional context / logs:**
As a workaround, setting entrypoint works, i.e.
```
docker run --rm --entrypoint /bin/kubectl rancher/k3s@sha256:5f7b660e6f2a6dd712350a8f1fd5ad6beafd62a06df5e2f46e432e6f151652fc
```

**Comment by brandond** [comment]
Did this work previously? If so, I'm curious when it broke. The main k3s binary for the docker image is different than the k3s artifact in GitHub, it does not have the same self-extracting behavior but the multi-call binary behavior should work when executed properly.

**Comment by brandond** [comment]
Yeah I don't think this ever worked. As far as I look back its always done the same:
```
brandond@dev01:~$ docker run --rm -it rancher/k3s:v1.19.15-k3s1 kubectl --help
Error: unknown command "kubectl" for "kubectl"
Run 'kubectl --help' for usage.
unknown command "kubectl" for "kubectl"
```

The true `k3s` subcommands - `agent`, `server`, and `etcd-snapshot` - work using `/bin/k3s` as the entrypoint, but the others do not.

I am a little confused why you would want to just run `kubectl` anyways - you should be able to exec into a running k3s container and run kubectl. Running kubectl directly as the entrypoint doesn't make much sense to me.

```
brandond@dev01:~$ docker exec -it k3s-server-1 kubectl get node
NAME           STATUS   ROLES                  AGE   VERSION
k3s-server-1   Ready    control-plane,master   17s   v1.28.2+k3s-5b6b9685
```
